client/src/contexts/user/UserContextProvider.js

```diff
@@ -0,0 +1,103 @@
+import * as React from "react";
+import { SIGN_UP, LOGIN, LOGOUT } from "../types";
+
+const initialState = {
+    isAuthenticated: false,
+    profile: {
+        _id: "",
+        email: "",
+        favoriteCuisine: [],
+        isChef: false,
+        primaryAddress: {},
+        profileImage: "",
+    },
+    token: null,
+};
+
+const UserReducer = (state, action) => {
+    switch (action.type) {
+        case SIGN_UP:
+        case LOGIN:
+            return {
+                ...state,
+                isAuthenticated: true,
+                profile: {
+                    _id: action.payload.user._id,
+                    email: action.payload.user.email,
+                    favoriteCuisine: action.payload.user.favoriteCuisine,
+                    isChef: action.payload.user.isChef,
+                    primaryAddress: {},
+                    profileImage: "",
```

Comment:
If a user has uploaded an image/added an address in the past, we want to override it - everything here should be set to the action.payload. Maybe test and see if you can do just `profile: action.payload.user`?

Reply:
I had that originally but then it was breaking when I was trying to display the email on the home page.  However after trying it again it works! Not sure what I changed to make it work but it is much cleaner